ablySpec/Auth.swift

```diff
@@ -599,5 +599,406 @@ class Auth : QuickSpec {
+
+                rest.auth.createTokenRequest(tokenParams, options: options, callback: { tokenRequest, error in
+                    expect(tokenRequest?.clientId).to(equal(expectedClientId))
+                    expect(tokenRequest?.mac).toNot(beNil())
+                    expect(tokenRequest?.nonce).toNot(beNil())
+                })
+            }
+
+        }
+
+        // RSA10
+        describe("authorise") {
+
+            // RSA10a
+            it("should create a token if needed and use it") {
+                let options = AblyTests.clientOptions(requestToken: true)
+                waitUntil(timeout: testTimeout) { done in
+                    // Client with Token
+                    let rest = ARTRest(options: options)
+                    publishTestMessage(rest, completion: { error in
+                        expect(error).to(beNil())
+                        expect(rest.auth.method).to(equal(ARTAuthMethod.Token))
+
+                        // Reuse the valid token
+                        rest.auth.authorise(nil, options: nil, callback: { tokenDetails, error in
+                            expect(rest.auth.method).to(equal(ARTAuthMethod.Token))
+                            expect(tokenDetails?.token).to(equal(options.token))
+
+                            publishTestMessage(rest, completion: { error in
+                                expect(error).to(beNil())
+                                done()
+                            })
+                        })
+                    })
+                }
+            }
+
+            // RSA10b
+            it("should supports all TokenParams and AuthOptions") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.authorise(ARTAuthTokenParams(), options: ARTAuthOptions(), callback: { tokenDetails, error in
+                        expect(error).to(beNil())
+                        done()
+                    })
+                }
+            }
+
+            // RSA10c
+            it("should create a new token when no token exists or current token has expired") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                let tokenParams = ARTAuthTokenParams()
+                tokenParams.ttl = 3.0 //Seconds
+
+                // FIXME: buffer of 15s for token expiry
+
+                var expiredToken: String?
+
+                // No token exists
+                expect(rest.auth.tokenDetails?.token).to(beNil())
+
+                // Created token
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.authorise(tokenParams, options: nil) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(tokenDetails).toNot(beNil())
+                        expect(tokenDetails?.token).toNot(beEmpty())
+
+                        if let token = tokenDetails?.token {
```

Comment:
I am not overly keen on this code in that we have what seems like a non-deterministic test.  Can we get read of this `if` condition and simply have one test process that succeeds every time?